File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        smallest_index =sys.maxsize
        for i in range(len(nums)):
            if i == self.sum_of_digit(nums[i]):
                smallest_index = i
                break
        if(smallest_index == sys.maxsize):
                return -1
        return smallest_index
    def sum_of_digit(self,n):
        total = 0
        while n > 0:
            total += n%10
            n //= 10
        return total

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int smallestIndex(int[] nums) {
        int smallest_index = Integer.MAX_VALUE;
        for(int i = 0 ;i < nums.length ; i++){
            if(i == sumofadigitindex(nums[i])){
            smallest_index = i;
            break;
            }
        }
        if(smallest_index == Integer.MAX_VALUE) return -1;
        return smallest_index;
    }

    int sumofadigitindex(int n){
        int sum = 0;
        while(n > 0){
         sum+= n%10;
         n = n/10;
        }
        return sum;
    }
}